crypto/Cargo.toml

```
@@ -34,10 +53,42 @@ parity-scale-codec = { workspace = true, features = ["derive", "full"] }
 serde = { workspace = true, features = ["derive"] }
 serde_with = { workspace = true, features = ["macros"] }
 hex = { workspace = true, features = ["alloc", "serde"] }
+# TODO: iroha_crypto no longer depends on openssl (did it ever?)
+# currently it's being used by iroha_client through attohttpc and iroha_cli through warp's tokeio-tunstenite
 openssl-sys = { version = "0.9.93", features = ["vendored"], optional = true }
-ursa = { workspace = true, optional = true }
 getset = { workspace = true }
 
+digest = { version = "0.10.7", optional = true }
+blake2 = { version = "0.10.6", optional = true }
+sha2 = { version = "0.10.8", optional = true }
+hkdf = { version = "0.12.3", optional = true }
+amcl = { version = "0.2.0", optional = true, default-features = false, features = ["secp256k1"] }
+amcl_wrapper = { version = "0.4.0", optional = true }
+
+signature = { version = "2.1.0", optional = true }
+ed25519-dalek = { version = "2.0.0", optional = true, features = ["rand_core"] }
+curve25519-dalek = { version = "4.1.1", optional = true }
+x25519-dalek = { version = "2.0.0", optional = true, features = ["static_secrets"] }
+
+rand = { version = "0.8.5", optional = true }
```

```suggestion
rand = { workspace = true, optional = true }
```

why not use from workspace?